### PR TITLE
[NG] Closes a Popover on ESC

### PR DESCRIPTION
**Pull request by adityarb88**
Fixes: #2062

Closes a popover when the `ESC` key is pressed.

Tested on Chrome

Note: For some reason `abstract-popover.ts` didn't have any tests. I have added tests related to this change in this PR. Will add the remaining tests in a separate PR.

Signed-off-by: Aditya Bhandari <adityab@vmware.com>

### CODE REVIEW

File: src/clr-angular/popover/common/abstract-popover.ts [hunk 1]
```diff
@@ -90,6 +92,13 @@ export abstract class AbstractPopover implements AfterViewChecked, OnDestroy {
         return this.ifOpenService.open ? false : true;
     }
 
+    @HostListener("keydown", ["$event"])
+    onKeyDown(event: KeyboardEvent) {
```
Comment (youdz) [inline src/clr-angular/popover/common/abstract-popover.ts]: This event listener is not on the document, so it only triggers if the key is pressed while focus is inside of the popover. Is that what we want?
I don't have an opinion, but whichever the decision is, please add a comment explaining it or linking to the corresponding conversation on GitHub.

Comment (youdz) [reply src/clr-angular/popover/common/abstract-popover.ts]: I don't believe dropdowns or tooltips should have a focus trap, unless I missed something. So you can't rely on that.
As I said, I'm fine either way, I just want the decision to be clear and explicitly commented. :)

Comment (adityarb88) [reply src/clr-angular/popover/common/abstract-popover.ts]: ok. i will check with @reddolan once. Thanks @youdz

**Comment by adityarb88** [comment]
@youdz: Per UX discussion, `dropdowns` and `tooltips` wont have `FocusTrap`, so that means we will have to add document level listeners like you suggested earlier. I have updated the PR to incorporate that change. Please take a look and let me know if this looks okay.

@hippee-lee: You might have to review the PR again.

**Pull request merged**